layers/data_service_layer/python/data_service.py

```diff
@@ -17,11 +18,60 @@ def __init__(self, table_name: str, dynamodb_resource=None):
         self.dynamodb = dynamodb_resource or boto3.resource('dynamodb')
         self.table = self.dynamodb.Table(table_name)
 
-    def get_markers(self) -> List[dict]:
+    def get_markers(self) -> List[LocationMarker]:
         """
         Retrieve all markers from the DynamoDB table.
 
         :return: A list of markers.
         :raises NotImplementedError: Indicates the method is not yet implemented.
         """
-        raise NotImplementedError("This method is not implemented yet.")
+        try: 
+            response = self.table.scan()
+            markers_data = response.get('Items', [])
+            markers = [LocationMarker.from_json(marker_data) for marker_data in markers_data]
+            logger.info(f"Retrieved {len(markers)} markers.")
+            return markers
+        except Exception as e:
+            logger.error(f"Error retrieving markers: {e}")
+            raise
+
+    def add_marker(self, location_marker: LocationMarker, mark_id):
```

Comment:
@PCBush we should generate the markerId inside this `add_marker` function and remove it from passed in parameters. For example

```
import uuid
unique_id = str(uuid.uuid4())
```

Reply:
Done